src/System.Management.Automation/engine/regex.cs

```diff
@@ -296,10 +302,12 @@ public static string Unescape(string pattern)
         {
             if (pattern == null)
             {
-                throw PSTraceSource.NewArgumentNullException("pattern");
+                throw PSTraceSource.NewArgumentNullException(nameof(pattern));
             }
 
-            char[] temp = new char[pattern.Length];
+            const int StackAllocThreshold = 256;
+            Span<char> temp = pattern.Length < StackAllocThreshold ? stackalloc char[StackAllocThreshold] : new char[pattern.Length];
```

Comment:
> `? stackalloc char[StackAllocThreshold]`

Should be `stackalloc char[pattern.Length]`

Reply:
Fixed.